.travis.yml

```
@@ -33,7 +33,7 @@ matrix:
     - node_js: "lts/*"
       env: REACT=16.8.3
     - node_js: "lts/*"
-      env: REACT=16.8 RENDERER=16.7
+      env: REACT=16.8 RENDERER=16.8
```

this is breaking the purpose of the test, which is to test react 16.8 with the 16.7 renderer.
```suggestion
      env: REACT=16.8 RENDERER=16.7
```

---

But  we need corresponding version of react and react-dom if there is version mismatch - hooks test will fail as writtern on react website one of reason for invalid hooks is
`Mismatching Versions of React and React DOM`

---

react and react-dom will both be 16.8 with this. `RENDERER` is about `react-test-renderer`, which doesn't necessarily have to match.

---

ok , then i can use `render, fireEvent, getByTestId` for updating the failing test cases?

---

The work that needs to be done to close the linked issue is to determine if the existing tests are valid - if so, the actual code needs to be fixed; or, if they're not valid, change them so they are valid.

---

thanks, I will look into it and update because in some test case we are simulating and clicking and expecting updated value but it won't `rerender` in the `older version` of `react-test-renderer` so we have to `manually call` the` rerender` function.
thanks again for replying

---

cool, maybe that's the fix we need then :-)

---

@ljharb  I was going through a lot of articles the react-test-renderer before 16.8.5 have problem with shallow rendering for hooks , as they won't re-render after setting State , and we are using same the same in Adaptors so shallow won't work  for version below 16.8.5,
now what we should do, modify the adaptor?

---

yes, ideally we'd modify the react 16 adapter to have fallback behavior for an older renderer version.

---

@AlokTakshak any interest in pursuing this fix?

---

I want to but i don't have any clue of how to fix it

---

My thinking is that the react 16 adapter would add some way (only for react-test-renderer prior to 16.8.5) to catch `useState` calls and force a rerender manually.